mesa/visualization/components/matplotlib.py

```diff
@@ -23,12 +25,42 @@ def SpaceMatplotlib(model, agent_portrayal, dependencies: list[any] | None = Non
     solara.FigureMatplotlib(space_fig, format="png", dependencies=dependencies)
 
 
+# matplotlib scatter does not allow for multiple shapes in one call
```

Comment:
It seems that Altair natively supports this: https://github.com/vega/altair/issues/1181.

Reply:
Altair space_drawer does not support continuous space. To have continuous space w/ multiple shapes altair must either support continuous space or matplotlib must support multiple shapes.

Reply:
I think we should do both, with the Altair one in a separate PR. In your code, it falls back to the default code branch when shape is not present, so it's fine.